sqld/src/lib.rs

```diff
@@ -228,6 +234,51 @@ fn check_fresh_db(path: &Path) -> bool {
     !path.join("wallog").exists()
 }
 
+fn validate_extensions(extensions_path: Option<PathBuf>) -> anyhow::Result<Vec<PathBuf>> {
+    let mut valid_extensions = vec![];
+    if let Some(ext_dir) = extensions_path {
+        let extensions_list = ext_dir.join("trusted.lst");
+
+        let file_contents = std::fs::read_to_string(&extensions_list)
+            .with_context(|| format!("can't read {}", &extensions_list.display()))?;
+
+        let extensions = file_contents.split_terminator('\n');
```

Comment:
nit: there's a `lines` methods on `str`: https://doc.rust-lang.org/std/primitive.str.html#method.lines

Reply:
Thanks. I thought that split_terminator would work better because it doesn't include the final newline in the file so I didn't have to filter. But it turns out it doesn't coalesce them, so I actually have to filter anyway (which I am not doing). Will update, using lines + filter.